### PR TITLE
OverviewCard: update to use DetailsCard change/onSave handling

### PR DESCRIPTION
**Pull request by sjd78**
After OverviewCard was developed and merged, some better techniques for managing VM edits were used in the DetailsCard #701.  This PR adopts the change handling and form style of the Details card to the OverviewCard.

There are no functional or visual changes to the card.

### CODE REVIEW

File: src/components/VmDetails/cards/OverviewCard/index.js [hunk 1]
```diff
@@ -76,42 +78,77 @@ class OverviewCard extends React.Component {
...
+
+    if (this.trackUpdates['description']) {
+      vmUpdates['description'] = stateVm.get('description')
+    }
+
+    // --- dispatch the save
+    //     saveChanges will add the result of the operation to the vm under the given
+    //     correlationId. So, when the vm prop changes, it can be checked and the edit
+    //     mode controlled based on the result of the dispatch/saga/api call.
+    console.info('saving changes to VM', vmUpdates.id, ', updates:', vmUpdates)
```
[the remark this answers is not in the document]
Comment (sjd78) [reply src/components/VmDetails/cards/OverviewCard/index.js]: I'm ok with taking out the log.

What kinds of things would you consider necessary or useful to keep in logging?  We can go back and remove/normalize logging in the core components so only things that add value get logged.  Maybe an email chain would be easier to discuss

File: src/components/VmDetails/cards/OverviewCard/index.js [hunk 2]
```diff
@@ -76,42 +78,77 @@ class OverviewCard extends React.Component {
...
+
+    let fieldUpdated
+    switch (fieldName) {
+      case 'name':
+        // TODO: add name validation?
+        updates = updates.set('name', value)
+        fieldUpdated = 'name'
+        break
+
+      case 'description':
```
[the remark this answers is not in the document]
Comment (sjd78) [reply src/components/VmDetails/cards/OverviewCard/index.js]: I intentionally left it as 2 explicit blocks to make future changes/validations a bit more obvious.

Actually, the edit of the VM Name on the legacy page actually will add a cloud-init hostname change depending on how the VM's template it setup.  I didn't realize that until I got to the end of the details card work.  So, I'm planning on opening an issue on it and see if it is something we want to put back in place.  

https://github.com/oVirt/ovirt-web-ui/blob/aff035eff23496834d34a7402497212459f86458/src/components/VmDialog/index.js#L241-L258

[the remark this answers is not in the document]
Comment (sjd78) [reply src/components/VmDetails/cards/OverviewCard/index.js]: See #803

File: src/components/VmDetails/cards/OverviewCard/index.js [hunk 3]
```diff
@@ -76,42 +78,77 @@ class OverviewCard extends React.Component {
...
+    }
+  }
+
   handleCardOnCancel () {
     this.setState({ isEditing: false, isDirty: false, correlationId: null, correlatedMessages: null })
     this.props.onEditChange(false)
   }
 
   handleCardOnSave () {
+    if (Object.keys(this.trackUpdates).length === 0) {
```
[the remark this answers is not in the document]
Comment (sjd78) [reply src/components/VmDetails/cards/OverviewCard/index.js]: True, but I'd prefer to keep the `=== 0` check, I find it just a bit easier to read quickly.

[the remark this answers is not in the document]
Comment (gregsheremeta) [reply src/components/VmDetails/cards/OverviewCard/index.js]: > I find it just a bit easier to read quickly
+1

**Pull request merged**